src/app/courses-dashboard/manage-course/manage-course.component.ts

```diff
@@ -60,14 +69,26 @@ export class ManageCourseComponent implements OnInit {
             name: course.name,
             description: course.description.trim(),
             length: course.length,
-            date: this.datePipe.transform(course.date, 'd MMM, yyyy'),
+            date: this.datePipe.transform(course.date, this.dateFormat),
             authors: course.authors
           });
         }
       });
     }
   }
 
+  onAuthorSearch(searchAuthorsString: string) {
+    if (searchAuthorsString.length !== 0) {
+      this.store.dispatch(getAuthors({ searchString: searchAuthorsString }));
+      this.authors$.subscribe((authors: Author[]) => {
+        console.log(authors);
+        this.isFind = authors.length > 0;
+      });
```

Comment:
if you're using store, ideally all this logic `this.authors$.subscribe((authors: Author[]) => {....` should be moved to selector